src/lib/cht-session.ts

```diff
@@ -95,6 +96,14 @@ export default class ChtSession {
     return setCookieHeader?.[0]?.split(';')
       .find((header: string) => header.startsWith(COUCH_AUTH_COOKIE_NAME));
   }
+
+  private static validateUserRole(userDoc: any, username: string): void {
+    const hasAuthorizedRole = (roles: string[]): boolean => _.intersection(AUTHORIZED_ROLES, roles).length > 0;
+
+    if (!hasAuthorizedRole(userDoc?.roles || [])) {
+      throw Error(`User ${username} role does not have the required permissions`);
```

Comment:
What happens when the user has the role but the role isn't configured with the right permissions?

Reply:
Here is my understanding. The role to perform actions on UMT is configured first then a user who can interact with UMT for instance `user_manager` is created with that role that has the permissions.

So a user who has role`['county_supervisor', 'user_manager']`. Such a user will be able to login to UMT because we will have added the check for `user_manager`

A user who only has `['county_supervisor'] `will not be able to login because they don't have the role with the permissions. 

Is your scenario still a concern from this perspective ?

Reply:
Yes because a cht admin could have configured the `user_manager` role without a permission like "can_create_users" or something like that. So even if they have the role, they don't have the required permissions. I'm not sure if it's possible to check if the role/user has the required permissions but ideally that would atleast strictly ensure one can't login. If not possible then this is fine

Reply:
I changed the approach. Now each config in the UMT will define which are the required permissions users need to be allowed access. 

The session now checks whether the user's roles have the permissions. The question now is, which permissions do we enforce for Echis ke ?

Reply:
They are defined in https://github.com/moh-kenya/config-echis-2.0/blob/9a140948070d0191990049dfb1fa9433165a99bb/app_settings/base_settings.json#L146

I don't think they need to be part of the uMT config as we already know beforehand what permissions are needed

Reply:
I've moved the permissions list to the service. The list of permissions needed to log into UMT will be defined in the service.